docs/instruments/ZJSONInstrument_v0.0.1.md

```diff
@@ -0,0 +1,1028 @@
+
+If the Z-JSON instrument is an original instrument, the value of this key is
+null.
+
+`meta.includeMetaDataFields`: Boolean. 
+
+True of false that the Z-JSON Instrument will include LORIS-specific
+metadata fields, such as 'Candidate Age', or 'Examiner', which
+calculates the candidate's age, and selects the examiner
+administering the instrument, respectively.  
+
+`meta.instrumentVersion`: Required string.
+
+The version of the instrument being described.
+
+It follows some convention as determined by the author of the instrument,
+starting with some representation of the number 1 and incrementing itself alongside any changes made to the `schema` object. It is self-contained and
+interpreted independently of the instrument's `shortName` or `longName`.
+
+`meta.lastUpdateOn`: String.
+
+The date at which small modifications, such as bug fixes, that do not justify
+a new version be created, was last made.
+                          
+`meta.longName`: Required object with IETF BCP 47 language tags as keys.
+
+The value of each key will be the long, human readable version of the instrument's name
+- its title - provided in every language that the instrument comes in.
+
+`meta.multilingual`: Boolean.
+
+True or false that the instrument is available in more than one language that is
+its `defaultLanguage`.
+
+`meta.previousVersion`: String.
+
+The previous version of the Z-JSON Instrument, if significant changes
+were made to justify creating this new version of the instrument.
+
+This may be redundant metadata but it may also be helpful in showing the
+lineage of existing instruments. If changes to an instrument is so
+significant that they no longer share the same lineage, `meta.derivedFrom`
+would be used to track changes. 
+
+`meta.schemaVersion`: Required string.
+
+The version of this Z-JSON Instrument Specification document that the
+instrument complies with. It follows the format of v*SpecVersion*-dev i.e. `v0.0.1-dev` if following this
+current version.
+
+`meta.shortName`: Required string.
+
+The unique name and identifier of the instrument, also the test name in the
+database.
+
+The name of the Z-JSON Instrument JSON file will follow the format *shortName_instrumentVersion*.
+
+## 2.2: Page Setup
+
+The `setup` key contains an array of objects that represent a page (or LORIS subtest) of an instrument. Each page
+object has the following format:
+
+```js
+{
+    "schema" : {
+        ...
+        "setup" : [
+            {
+                "name" : string,
+                "type" : "page",
+                "description" : string,
+                "order": [
+                  "nameOfField",
+                      ...
+                ]
+            }
+        ]
+    }
+}
+```
+
+`setup.name`: String.
+              The name and ID of the page object.
+
+`setup.typ`: Required string with value "page".
+
+`setup.description`: Required string. The label or title of the page that will be rendered on the front-end as a link to the
```

Comment:
so this is my intention: if `multilingual`: false, then all `description` keys have value strings that are the description text. if `multilingual`: true, then `description` keys have value object with language tags as keys and description strings as values.

JSON-LD only needs language tags if there are multiple languages. otherwise, the string suffices and you define a "default language" in the context of the JSON-LD to compensate. I'm trying to imitate this so its easier to parse.

I guess I should define this earlier so it's clear that `description` can be either a string or an object depending on `multilingual`. what do you think about keeping the multilingual key in the meta component after all?

Reply:
I still don't like it. Types shouldn't be context-sensitive.